drivers/entropy/entropy_stm32.c

```diff
@@ -361,6 +361,18 @@ static int entropy_stm32_rng_get_entropy_isr(const struct device *dev,
 		irq_disable(IRQN);
 		irq_unlock(key);
 
+		/* do not proceed if a Seed error occurred */
+		if (LL_RNG_IsActiveFlag_SECS(entropy_stm32_rng_data.rng) ||
+			LL_RNG_IsActiveFlag_SEIS(entropy_stm32_rng_data.rng)) {
+
+			(void)random_byte_get(); /* this will recover the error */
+			/* restore irq as we enter */
+			if (irq_enabled) {
+				irq_enable(IRQN);
+			}
+			return 0; /* return cnt is null : no random data available */
+		}
```

Comment:
wouldn't be better to return some error code here instead of 0 ?

Reply:
yes the function could  `return -EIO; ` likle in the _recover_seed_error_ function, on the other hand, the "number – of bytes filled with entropy" is actually 0.